storage/client/client.go

```diff
@@ -1114,6 +1117,23 @@ func (c *StorageClient) RuntimeEvents(ctx context.Context, p apiTypes.GetRuntime
 		); err != nil {
 			return nil, wrapError(err)
 		}
+
+		// Re-encode EVM log params of integer type as strings, for two reasons:
+		//  - If a value is a large enough int, it will be (re)marshaled as a JSON float
+		//    somewhere in the stack.
+		//  - For JS clients, big ints are best represented as strings. This is also
+		//    consistent with how the rest of the API represents the BigInt type.
+		if e.EvmLogParams != nil {
+			for i := range *e.EvmLogParams {
+				param := &(*e.EvmLogParams)[i]
+				// param.Value is a json.RawMessage, so "marshaling" it just returns
+				// the same raw JSON-encoded bytes that came from the DB.
+				JSONValue, _ := param.Value.MarshalJSON()
+				if digitTest.MatchString(string(JSONValue)) {
```

Comment:
Since `json.RawMessage` is []byte, it looks like we can maybe just use `param.Value` directly instead of calling [MarshalJSON()](https://cs.opensource.google/go/go/+/refs/tags/go1.20.2:src/encoding/json/stream.go;l=265)?

ie, something like
```
MatchString(string(param.Value))
```

Reply:
We could, and it works fine, I just tried. However this relies on what is technically an internal implementation detail of the `json` library. I'd rather preserve the abstraction boundaries and use the intended API for converting to []byte (and eventually string).